app/controllers/post.js

```diff
@@ -1,16 +1,89 @@
 const jwt = require('jsonwebtoken')
+const PostModel = require('../models/Post')
+const HANDLER = require('../utils/response-helper')
+const STATUS = require('../utils/status-codes')
+const helper = require('../utils/uploader')
 
 module.exports = {
-  create: function (req, res, next) {
-    userModel.create({ name: req.body.name, email: req.body.email, password: req.body.password }, function (err, result) {
-      if (err) {
-        next(err)
-      } else {
-        res.json({ status: 'success', message: 'User added successfully!!!', data: null })
+  create: async (req, res, next) => {
+    const post = new PostModel(req.body)
+    if (req.file) {
+      helper.mapToDb(req, post)
+    }
+    try {
+      await post.save()
+      res.status(STATUS.CREATED).json({ post })
+    } catch (error) {
+      HANDLER.handleError(res, error)
+    }
+  },
+  delete: async (req, res, next) => {
+    const { id } = req.params
+    console.log(id)
+    try {
+      const post = await PostModel.findByIdAndRemove(id)
+      if (!post) {
+        return res.status(STATUS.NOT_FOUND).json({ message: 'No post exists' })
       }
+      res.status(STATUS.OK).json({ post: post, message: 'Deleted the post' })
+    } catch (error) {
+      HANDLER.handleError(res, error)
+    }
+  },
+  updatePost: async (req, res, next) => {
+    const { id } = req.params
+    console.log(id)
+    const updates = Object.keys(req.body)
+    const allowedUpdates = ['content', 'imgUrl']
+    const isValidOperation = updates.every((update) => {
+      return allowedUpdates.includes(update)
     })
+
+    if (!isValidOperation) {
+      return res.status(STATUS.BAD_REQUEST).json({ message: 'Invalid Update' })
+    }
+    try {
+      const post = await PostModel.findById(id)
+      console.log(post)
+      updates.forEach(update => {
+        post[update] = req.body[update]
+      })
+      if (req.file) {
+        helper.mapToDb(req, post)
+      }
+      await post.save()
+      res.status(STATUS.UPDATED).json({ post: post })
+    } catch (error) {
+      HANDLER.handleError(res, error)
+    }
+  },
+  getPostById: async (req, res, next) => {
+    const { id } = req.params
+    try {
+      const post = await PostModel.findById(id)
+      if (!post) {
+        return res.status(STATUS.NOT_FOUND).json({ error: 'Post not found' })
+      }
+      res.status(STATUS.OK).json({ post: post })
+    } catch (error) {
+      HANDLER.handleError(res, error)
+    }
+  },
+  getAllPost: async (req, res, next) => {
+    try {
+      await PostModel.find({}, (error, posts) => {
+        if (error) {
+          return res.status(STATUS.BAD_REQUEST).json({ error: error })
+        } if (!posts.length) {
+          return res.status(STATUS.NOT_FOUND).json({ message: 'Item not found' })
+        }
+        return res.status(STATUS.OK).json({ success: true, data: posts })
+      })
+    } catch (error) {
+      HANDLER.handleError(res, error)
+    }
   },
   authenticate: function (req, res, next) {
```

Comment:
why do we have this function here?

Reply:
That was there earlier, I did not touch it anyway. I will remove it, seems like it is of no use there.